### PR TITLE
[WIP] Get units fixes

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

File: pootle/apps/pootle_store/unit/filters.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    def search(self, text, sfields, exact=False):
+        if exact:
+            words = [text]
+        else:
+            words = text.split()
+        result = self.qs.none()
+        for k, v in self.search_mappings.items():
+            if k in sfields:
+                sfields += v
+        for k in filter(lambda x: x in self.search_fields, sfields):
```
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/filters.py]: Codacy Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1704058174/issues/source?bid=1990595&fileBranchId=2994326#l44)

File: tests/forms/unit.py [hunk 2]
```diff
@@ -139,3 +145,285 @@ def test_unit_state():
...
+        assert search._order_by == sorted_qs.query.order_by[0]
+        assert search.order_by == search._order_by.strip("-")
+    else:
+        assert search._order_by is None
+        assert search.order_by == "pk"
+
+    lte = (
+        cleaned["sort_by"]
+        and cleaned["sort_by"].startswith("-"))
+    if lte:
```
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/1704058323/issues/source?bid=1990595&fileBranchId=2994326#l389)

File: tests/forms/unit.py [hunk 3]
```diff
@@ -139,3 +145,285 @@ def test_unit_state():
...
+        search.compare_with == "lte"
+    else:
```
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/1704058323/issues/source?bid=1990595&fileBranchId=2994326#l391)

File: tests/forms/unit.py [hunk 4]
```diff
@@ -139,3 +145,285 @@ def test_unit_state():
...
+
+    return uids, units
+
+
+def _test_form_filter(form, result):
+    uids, expected = _calculate_results(form)
+    unit_count = 0
+    for group in result["unitGroups"]:
+        for result_store, info in group.items():
+            for result_unit in info['units']:
```
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/1704058323/issues/source?bid=1990595&fileBranchId=2994326#l234)

File: pootle/apps/pootle_store/unit/filters.py (hunk 1 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/filters.py]: Codacy Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1761713011/issues/source?bid=1990595&fileBranchId=2994326#l44)

Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/filters.py]: Codacy Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1770470975/issues/source?bid=1990595&fileBranchId=2994326#l44)

Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/filters.py]: Codacy Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1770470975/issues/source?bid=1990595&fileBranchId=2994326#l44)

Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/filters.py]: Codacy Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1792218155/issues/source?bid=1990595&fileBranchId=2994326#l44)

File: pootle/apps/pootle_store/unit/search.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+        kwargs = dict(
+            language_code=self.kwa["language"],
+            project_code=self.kwa["project"],
+            dir_path=self.kwa["dir_path"],
+            filename=self.kwa["filename"])
+
+        if self.kwa.get("pootle_path", None):
+            return Unit.objects.get_translatable(
+                self.request_user, **kwargs)
```
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/search.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/1996784469/issues/source?bid=1990595&fileBranchId=2994326#l45)

File: pootle/apps/pootle_store/unit/filters.py (hunk 1 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/filters.py]: Codacy Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1996784465/issues/source?bid=1990595&fileBranchId=2994326#l44)

File: pytest_pootle/fixtures/forms.py [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    params["cleaned_data"].update(params.get("data", {}))
+
+    if "get" in params:
+        if "uids" in params["get"]:
+            params["cleaned_data"]["uids"] = [params["get"]["uids"]()]
+    params["cleaned_data"]['path'] = unicode(params["get"]["path"])
+    params["cleaned_data"]['pootle_path'] = params["cleaned_data"]['path']
+
+    limit = params.get("limit", None)
+
```
Comment (dwaynebailey) [inline pytest_pootle/fixtures/forms.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/1996784650/issues/source?bid=1990595&fileBranchId=2994326#l206)

File: pootle/apps/pootle_store/unit/search.py (hunk 5 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/search.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/1998901215/issues/source?bid=1990595&fileBranchId=2994326#l45)

File: pytest_pootle/fixtures/forms.py (hunk 6 above)
Comment (dwaynebailey) [inline pytest_pootle/fixtures/forms.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/1998901782/issues/source?bid=1990595&fileBranchId=2994326#l206)

File: pootle/apps/pootle_store/unit/search.py (hunk 5 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/search.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/1998901215/issues/source?bid=1990595&fileBranchId=2994326#l45)

File: pytest_pootle/fixtures/forms.py (hunk 6 above)
Comment (dwaynebailey) [inline pytest_pootle/fixtures/forms.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/1998901782/issues/source?bid=1990595&fileBranchId=2994326#l206)

File: tests/forms/unit.py [hunk 7]
```diff
@@ -145,3 +151,285 @@ def test_unit_state():
...
```
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l240)

File: tests/forms/unit.py [hunk 8]
```diff
@@ -145,3 +151,285 @@ def test_unit_state():
...
+    else:
+        assert search._order_by is None
+        assert search.order_by == "pk"
+
+    lte = (
+        cleaned["sort_by"]
+        and cleaned["sort_by"].startswith("-"))
+    if lte:
+        search.compare_with == "lte"
+    else:
```
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l397)

File: tests/forms/unit.py [hunk 9]
```diff
@@ -145,3 +151,285 @@ def test_unit_state():
...
+        assert search._order_by == sorted_qs.query.order_by[0]
+        assert search.order_by == search._order_by.strip("-")
+    else:
+        assert search._order_by is None
+        assert search.order_by == "pk"
+
+    lte = (
+        cleaned["sort_by"]
+        and cleaned["sort_by"].startswith("-"))
+    if lte:
```
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l395)

File: pytest_pootle/fixtures/forms.py [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
```
Comment (dwaynebailey) [inline pytest_pootle/fixtures/forms.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/2000838858/issues/source?bid=1990595&fileBranchId=2994326#l203)

File: pootle/apps/pootle_store/unit/search.py (hunk 5 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/search.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/2000838726/issues/source?bid=1990595&fileBranchId=2994326#l45)

File: tests/forms/unit.py (hunk 7 above)
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l240)

File: tests/forms/unit.py (hunk 8 above)
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l397)

File: tests/forms/unit.py (hunk 9 above)
Comment (dwaynebailey) [inline tests/forms/unit.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l395)

File: pytest_pootle/fixtures/forms.py (hunk 10 above)
Comment (dwaynebailey) [inline pytest_pootle/fixtures/forms.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/2000838858/issues/source?bid=1990595&fileBranchId=2994326#l203)

File: pootle/apps/pootle_store/unit/search.py (hunk 5 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_store/unit/search.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/dwayne/pootle/file/2000838726/issues/source?bid=1990595&fileBranchId=2994326#l45)

**Comment by phlax** [comment]
this has been dealt with in other PRs

**Pull request closed without merging**